src/test/java/bio/terra/service/dataset/DatasetConnectedTest.java

```diff
@@ -507,8 +507,128 @@ public void testRepeatedSoftDelete() throws Exception {
+        List<String> softDeleteRowIds2 = getRowIdsFromBQTable(summaryModel.getName(), internalSoftDeleteTableName);
+        assertEquals("Size of soft delete table is still 1", 1, softDeleteRowIds2.size());
+        assertTrue("Soft deleted row id is still in soft delete table", softDeleteRowIds2.contains(softDeleteRowId));
+
+        // delete the dataset and check that it succeeds
+        connectedOperations.deleteTestDatasetAndCleanup(summaryModel.getId());
+
+        // try to fetch the dataset again and confirm nothing is returned
+        connectedOperations.getDatasetExpectError(summaryModel.getId(), HttpStatus.NOT_FOUND);
+    }
+
+    @Test
+    public void testRepeatedSoftDeleteHitThatQuota() throws Exception {
+        // load a CSV file that contains the table rows to load into the test bucket
+        String resourceFileName = "snapshot-test-dataset-data.csv";
+        String dirInCloud = "scratch/testRepeatedSoftDelete/" + UUID.randomUUID().toString();
+        BlobInfo ingestTableBlob = BlobInfo
+            .newBuilder(testConfig.getIngestbucket(), dirInCloud + "/" + resourceFileName)
+            .build();
+        Storage storage = StorageOptions.getDefaultInstance().getService();
+        storage.create(ingestTableBlob,
+            IOUtils.toByteArray(getClass().getClassLoader().getResource(resourceFileName)));
+        String tableIngestInputFilePath =
+            "gs://" + testConfig.getIngestbucket() + "/" + dirInCloud + "/" + resourceFileName;
+
+        // ingest the table
+        String tableName = "thetable";
+        IngestRequestModel ingestRequest = new IngestRequestModel()
+            .table(tableName)
+            .format(IngestRequestModel.FormatEnum.CSV)
+            .csvSkipLeadingRows(1)
+            .path(tableIngestInputFilePath);
+        connectedOperations.ingestTableSuccess(summaryModel.getId(), ingestRequest);
+
+        // make sure the JSON file gets cleaned up on test teardown
+        connectedOperations.addScratchFile(dirInCloud + "/" + resourceFileName);
+
+        // load a CSV file that contains the table rows to soft delete into the test bucket
+        String softDeleteRowId = "8c52c63e-8d9f-4cfc-82d0-0f916b2404c1";
+        List<String> softDeleteRowIds = new ArrayList<>();
+        softDeleteRowIds.add(softDeleteRowId); // add the same rowid twice
+        softDeleteRowIds.add(softDeleteRowId);
+        DataDeletionRequest softDeleteRequest = uploadInputFileAndBuildSoftDeleteRequest(
+            dirInCloud, "testRepeatedSoftDelete.csv", tableName, softDeleteRowIds);
+
+        // make the soft delete request and wait for it to return
         connectedOperations.softDeleteSuccess(summaryModel.getId(), softDeleteRequest);
 
+        // check that the size of the live table matches what we expect
+        List<String> liveTableRowIds1 = getRowIdsFromBQTable(summaryModel.getName(), tableName);
+        assertEquals("Size of live table is 3", 3, liveTableRowIds1.size());
```

Comment:
We can tell that the raw table (joined with the external table in the erring query) is not empty because the live view is based off of it.